src/liballoc/collections/vec_deque.rs

```diff
@@ -2809,7 +2809,23 @@ impl<'a, T> IntoIterator for &'a mut VecDeque<T> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<A> Extend<A> for VecDeque<A> {
     fn extend<T: IntoIterator<Item = A>>(&mut self, iter: T) {
-        iter.into_iter().for_each(move |elt| self.push_back(elt));
+        // This function should be the moral equivalent of:
+        //
+        //      for item in iter.into_iter() {
+        //          self.push_back(item);
+        //      }
+        let mut iter = iter.into_iter();
+        while let Some(element) = iter.next() {
+            let len = self.len();
+            if len == self.capacity() {
```

Comment:
You don't seem to use `len` a second time, you could directly replace it

Reply:
Yes, that was the result of momentary confusion and I'm just going to squash in the fix. Thanks!